### PR TITLE
Final Project Submission

### PR DESCRIPTION
**Pull request by colter107**
Final submission for final project

### CODE REVIEW

**Review by jwilcox76 — APPROVED** [review]
This merge is massive.. just like Ninja's low taper fade meme

**Comment by tvannoy** [comment]
> Ninja's low taper fade meme

TIL about the low taper fade meme

File: linux/servo-pwm/servo.c [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+#define duty_cycle_OFFSET 0x0
+#define period_reg_OFFSET 0x04
+
+#define SPAN 16
+
+
+/**
+* struct servo_dev - Private <component> device struct.
+* @base_addr: Pointer to the component's base address
+* @<reg_name>: <reg_purpose>
```
Comment (tvannoy) [inline linux/servo-pwm/servo.c:21]: outdated comment -5%

**Comment by tvannoy** [comment]
# Grading

- rgb demo: 100%
- proposal demo: 75%
- code: 80%
- documentation: 90%
- **total: 86.25%**

## proposal demo

one proposal worked, other partly worked

## code

-5 incorrect span, -5 could have used smaller address width, -5 formatting, -5 outdated comments

## documentation

-5 no conclusion, -5 no top-level readme

**Pull request merged**